### ISSUE TITLE
Fix the Arcs Live Deployment

### ISSUE THREAD

**Issue by alxrsngrtn** [PolymerLabs/arcs#4106]
Travis has been having errors deploying to the arcs live repo (temporary solution was to postpone this process -- #4105). 

```
 23903 files changed, 35493 insertions(+), 4812109 deletions(-)
warning: inexact rename detection was skipped due to too many files.
warning: you may want to set your diff.renameLimit variable to at least 21924 and retry the command.
Doing the git push (workdir: /tmp/d20191119-11200-how5ul/work)...
Couldn't push the build to github.com/PolymerLabs/arcs-live.git:master
```

It seems like the deploy process needs to be refactored to only push arcs live, and not extra content.

**Comment by sjmiles** [comment]
> It seems like the deploy process needs to be refactored to only push arcs live, and not extra content.

It probably would simplify matters to push a proper deployment, but I'd like to have a root cause of this actual problem. Knowledge is power.

**Comment by shaper** [comment]
Original change is https://github.com/PolymerLabs/arcs/pull/4008

**Comment by shaper** [comment]
@piotrswigon is looking at rolling arcs-live back to a build before CI broke it, to unblock those who depend on it. He may not own the fix but I figured even transient ownership of a P1 is worthwhile.

**Comment by piotrswigon** [comment]
arcs live is back to a working version, but updating it is disabled.

**Comment by shaper** [comment]
I talked with @alxrsngrtn he is going to look at this today.

**Comment by shaper** [comment]
@sjmiles and I think a first step is to look at filtering the appropriate bazel artifacts (`bin`?) following the ignore filtering we already do, in `.travis.yml`, see ex:

```
#  # Travis pages deploys skips artifacts from .gitignore. We need some of those, so
#  # remove them from .gitignore.
#  - cd ${TRAVIS_BUILD_DIR} && sed -i.sedbackup -e '/^shells\/env\/build$/d' .gitignore
```

an excerpt from the disabling change you landed earlier.

**Comment by alxrsngrtn** [comment]
I'm going to start working on this now. I'll report along the way or thereafter what's going down.

**Comment by alxrsngrtn** [comment]
Here's a demo of what a minimal* set of files to push to the arcs-live repo could look like. 

https://github.com/alxrsngrtn/arcs-live/tree/gh-pages

I'm serving a version of this on my personal GH pages account. Feel free to poke around and see if we still get full functionality. 

https://alexrosengarten.com/arcs-live/

**Comment by alxrsngrtn** [comment]
It seems like the old travis script was pushing `node_modules` (with modification) to arcs-live. Does anyone know why? 
``` 
# Temporarily push node_modules to arcs-live,
  # with the exception of exe files as Github complains about them.
  - cd ${TRAVIS_BUILD_DIR} && sed -i.sedbackup -e 's/^\*\*\/node_modules$/\*\*\/node_modules\/\*\*\/\*\.exe/' .gitignore
  - cd ${TRAVIS_BUILD_DIR} && rm -f server/node_modules/arcs
```

**Comment by shaper** [comment]
It's still failing to deploy due, I think, to the latest attempt's `mv` reporting an error due to failure to move '.' and '..'. I can't put together another PR right now. I'll look at it again in the morning unless someone else has gotten to it first. @alxrsngrtn

**Comment by shaper** [comment]
The push to arcs-live is still broken for some reason. arcs-live is serving from gh-pages branch (at older working commit) currently. Suggest debugging the deploy further in a PR (take out the conditional on master-commit only for deploy) as a next step.

**Comment by alxrsngrtn** [comment]
This issue should be solved via the PRs linked to the issue.

**Comment by alxrsngrtn** [comment]
Now with #4164, the arcs-live deploy is fixed.
